### PR TITLE
String / it-academy.by / M-JC1-69-21 / Bozhen Davydov / Done tasks 1 - 18

### PR DESCRIPTION
**Pull request by davydovbv**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ java {
 }
 
 dependencies {
+    implementation 'org.jetbrains:annotations:20.1.0'
```
Comment (rakovets) [inline build.gradle:15]: What is it???

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringUtil {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:11]: Remove blank line

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    public String glue(String str1, String str2) {
+        return str1.concat(str2);
+    }
+
+    public  int  findIndexOfSymbol(char ch, String str) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:16]: Violations of the Code of the Convention

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        result = new String(container);
+
+        return result;
+    }
+
+    public String replaceEmoji(String str1, String str2, String str3) {
+        return  str1.replace(str2,str3);
+    }
+
+    public boolean correctBegin( String text, String word) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:41]: Violations of the Code of the Convention

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        String result;
+        char[] container = new char[indexEnd - indexBegin];
+        str.getChars(indexBegin, indexEnd, container, 0);
+        result = new String(container);
+
+        return result;
+    }
+
+    public String replaceEmoji(String str1, String str2, String str3) {
+        return  str1.replace(str2,str3);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:38]: Violations of the Code of the Convention

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+        Matcher matcher = pattern.matcher(text);
+        while (matcher.find()) {
+            countOfPunctuationMarks++;
+        }
+
+        return countOfPunctuationMarks;
+    }
+
+    public boolean palindromChekOut(String text) {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:68]: Violations of the Code of the Convention

**Review by rakovets — CHANGES_REQUESTED** [review]
Please, fix all violations of the Code of the Convention.
After I will to do code review

File: build.gradle (hunk 1 above)
Comment (davydovbv) [reply build.gradle:15]: Have no idea what is this)

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+        results[1]= timeForStringBuilder;
+
+        return results;
+    }
+
+
+
+}
+
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:209]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+        long[] results = new long[2];
+        results[0] = timeForString;
+        results[1]= timeForStringBuilder;
+
+        return results;
+    }
+
+
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:206]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+        for (char firstWordChar : firstWordChars) {
+            int count = 0;
+            for (int i = 0; i < secondWordChars.length; i++) {
+                if(firstWordChar == secondWordChars[i]) {
+                    count ++;
+                    secondWordChars[i] = ' ';
+                    break;
+                }
+            }
+            if (count == 0){
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:132]: "Violation: There isn't separating white spaces for curly brake
Solution: Add white spaces"

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+    public String[] splitText(String  text, int numberSymbols) {
+        String[] splitText;
+        int j = 0;
+        if (text.length() % numberSymbols != 0) {
+            splitText = new String[text.length() / numberSymbols+1];
+        } else {
+            splitText = new String[text.length() / numberSymbols];
+        }
+
+        for(int i = 0; i < text.length() ; i += numberSymbols) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:75]:
```suggestion
        for(int i = 0; i < text.length(); i += numberSymbols) {
```

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+
+    public boolean palindromChekOut(String text) {
+        return text.toLowerCase().replaceAll("[\\pP\\s]", "").equals(new StringBuilder(text.toLowerCase().replaceAll("[\\pP\\s]", "")).reverse().toString());
+    }
+
+    public String[] splitText(String  text, int numberSymbols) {
+        String[] splitText;
+        int j = 0;
+        if (text.length() % numberSymbols != 0) {
+            splitText = new String[text.length() / numberSymbols+1];
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:70]:
```suggestion
            splitText = new String[text.length() / numberSymbols + 1];
```

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix the following problems

**Pull request closed without merging**